R/competing_hazards.R

```diff
@@ -16,22 +16,25 @@ CompetingOutcome <- R6::R6Class(
 
       self$target <- individual::Bitset$new(size)
       self$rates <- NULL
+      self$args <- NULL
     },
-    set_rates = function(target, rates){
+    set_rates = function(target, rates, ...){
       stopifnot(target$size() == length(rates))
-
       self$target$copy_from(target)
       self$rates <- rates
+      self$args <- list(...)
     },
     execute = function(t, target){
-      private$targeted_process(t, target)
+      do.call(private$targeted_process, list(t, target, self$args))
```

Comment:
Oh, I think it's most intuitive to call `targeted_process` with the extra arguments in the same way they were passed to `$set_rates`. i.e.

`outcome$set_rates(rates, a, b)`

should call the process with:

`targeted_process(target, a, b)` <- this version preserves the structure of my extra arguments

not

`targeted_process(target, list(a, b))` <- this version turns my extra arguments into a list.

```suggestion
      do.call(private$targeted_process, c((t, target), self$args))
```